README.md

```diff
@@ -5,9 +5,32 @@
 [![crates.io](https://img.shields.io/crates/v/narrow.svg)](https://crates.io/crates/narrow)
 [![docs.rs](https://docs.rs/narrow/badge.svg)](https://docs.rs/narrow)
 
-An experimental (work-in-progress) implementation of [Apache Arrow](https://arrow.apache.org).
+An experimental (work-in-progress) statically typed implementation of [Apache Arrow](https://arrow.apache.org).
 
-This crate provides types to support reading and writing instances of abstract data types in Arrow's in-memory data structures.
+This crate provides methods to automatically generate types to support reading and writing instances of abstract data types in Arrow's in-memory data structures.
+
+## Why
+
+- The [arrow](https://docs.rs/arrow) crate provides APIs that mostly make sense when array types are only known at runtime, however, for applications with [types known at compile-time](https://docs.rs/arrow/latest/arrow/#type-erasure--trait-objects), using [trait objects](https://doc.rust-lang.org/book/ch17-02-trait-objects.html) and [downcasting](https://docs.rs/arrow/latest/arrow/array/fn.downcast_array.html) is not ergonomic.
```

Comment:
How about:
```suggestion
- The [arrow](https://docs.rs/arrow) crate provides APIs that make sense when the array types are only known at run-time. Many of its [APIs](https://docs.rs/arrow/latest/arrow/#type-erasure--trait-objects) require the use of [trait objects](https://doc.rust-lang.org/book/ch17-02-trait-objects.html) and [downcasting](https://docs.rs/arrow/latest/arrow/array/fn.downcast_array.html). However, for many applications, types are known at compile-time. This makes working with the `arrow-rs` APIs feel less ergonomic, and the use of trait objects and downcasting is less efficient.
```

And then given this context, add another sentence explaining the alternative approach that `narrow` provides.

Reply:
I tried to explain the alternative approach below, [here](https://github.com/mbrobbel/narrow/pull/177/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R17): 
`There are [other crates](https://crates.io/search?q=arrow%20derive&sort=relevance) that aim to prevent users from having to maintain array builder code by providing derive macros. These builders typically produce type-erased arrays, whereas this crate only provides fully statically typed arrays.`

I mostly agree with your suggestion, however, I think the efficiency argument is weak without data, so I suggest we drop that:
```suggestion
- The [`arrow`](https://docs.rs/arrow) crate provides APIs that make sense when the array types are only known at run-time. Many of its [APIs](https://docs.rs/arrow/latest/arrow/#type-erasure--trait-objects) require the use of [trait objects](https://doc.rust-lang.org/book/ch17-02-trait-objects.html) and [downcasting](https://docs.rs/arrow/latest/arrow/array/fn.downcast_array.html). However, for applications where types are known at compile-time, these APIs are not ergonomic.
```

Reply:
I agree that it would be best to defer any statements about efficiency until we have benchmarks. It's definitely not the main "why".